augur/dates.py

```diff
@@ -1,73 +1,271 @@
-    """Wraps numeric_date() for argparse usage.
+def any_to_numeric_type_min(date_in:Any):
+    """Get the numeric date from any supported date format, taking the minimum possible value if ambiguous.
 
-    This raises an ArgumentTypeError, otherwise the custom exception message won't be shown in console output due to:
+    This function is intended to be used as the `type` parameter in `argparse.ArgumentParser.add_argument()`
+
+    This raises an ArgumentTypeError from InvalidDateFormat exceptions, otherwise the custom exception message won't be shown in console output due to:
     https://github.com/python/cpython/blob/5c4d1f6e0e192653560ae2941a6677fbf4fbd1f2/Lib/argparse.py#L2503-L2513
+
+    >>> round(any_to_numeric_type_min(2018), 3)
+    2018.001
     """
     try:
-        return numeric_date(date)
-    except ValueError as e:
+        return any_to_numeric(date_in, ambiguity_resolver='min')
+    except InvalidDateFormat as e:
         raise argparse.ArgumentTypeError(str(e)) from e
 
+
+def any_to_numeric_type_max(date_in:Any):
+    """Get the numeric date from any supported date format, taking the maximum possible value if ambiguous.
+
+    This function is intended to be used as the `type` parameter in `argparse.ArgumentParser.add_argument()`
+
+    This raises an ArgumentTypeError from InvalidDateFormat exceptions, otherwise the custom exception message won't be shown in console output due to:
+    https://github.com/python/cpython/blob/5c4d1f6e0e192653560ae2941a6677fbf4fbd1f2/Lib/argparse.py#L2503-L2513
+
+    >>> round(any_to_numeric_type_max(2018), 3)
+    2018.999
+    """
+    try:
+        return any_to_numeric(date_in, ambiguity_resolver='max')
+    except InvalidDateFormat as e:
+        raise argparse.ArgumentTypeError(str(e)) from e
+
+
+### date_to_numeric logic ###
+# copied from treetime.utils.numeric_date
+# simplified+cached for speed
+
+from calendar import isleap
+def date_to_numeric(date:datetime.date):
+    """Return the numeric date representation of a datetime.date."""
+    days_in_year = 366 if isleap(date.year) else 365
+    return date.year + (date.timetuple().tm_yday-0.5) / days_in_year
+
+
+@lru_cache(maxsize=CACHE_SIZE)
+def date_to_numeric_capped(date:datetime.date, max_numeric:float=None):
+    """Return the numeric date representation of a datetime.date, capped at a maximum numeric value.
+
+    Parameters
+    ----------
+    date
+        Date to convert.
+    max_numeric
+        Maximum numeric value to return. Default is current date.
+
+    >>> round(date_to_numeric_capped(datetime.date(2017, 1, 1), max_numeric=2018.0), 3)
+    2017.001
+    >>> round(date_to_numeric_capped(datetime.date(2019, 1, 1), max_numeric=2018.0), 3)
+    2018.0
+    """
+    if max_numeric is None:
+        max_numeric = date_to_numeric(datetime.date.today())
+    date_numeric = date_to_numeric(date)
+    if date_numeric > max_numeric:
+        date_numeric = max_numeric
+    return date_numeric
```

Comment:
This can be simplified to:

```suggestion
return min(date_numeric, max_numeric)
```

Reply:
Done: dc754b7a